cpmpy/transformations/flatten_model.py

```diff
@@ -436,7 +436,7 @@ def normalized_boolexpr(expr):
 
                 # this is a reified constraint, so lhs must be var too to be in normal form
                 (lhs, lcons) = get_or_make_var(lexpr)
-                if expr.name == '!=':
+                if expr.name == '!=' and is_boolexpr(rvar):
```

Comment:
You can just do `rvar.is_bool()` which will be more efficient compared to checking `is_boolexpr`.
`rvar` is guaranteed to be a variable at this point.